sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1219,7 +1241,8 @@ class Analyzer(
 
       // Make sure a plan's expressions do not contain outer references
       def failOnOuterReference(p: LogicalPlan): Unit = {
-        if (p.expressions.exists(containsOuter)) {
+        p.expressions.foreach(checkMixedReferencesInsideAggregation)
+        if (!p.isInstanceOf[Filter] && p.expressions.exists(containsOuter)) {
```

Comment:
I think we should put the check of `containsOuter` before `checkMixedReferencesInsideAggregation`?